File: src/main/java/io/zeitwert/fm/app/adapter/api/rest/dto/UserInfoResponse.java
package io.zeitwert.fm.app.adapter.api.rest.dto;

import io.zeitwert.ddd.enums.adapter.api.jsonapi.dto.EnumeratedDto;
import io.zeitwert.ddd.oe.adapter.api.jsonapi.dto.ObjTenantDto;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class UserInfoResponse {

	private Integer id;
	private String name;
	private String email;
	private ObjTenantDto tenant;
	private EnumeratedDto role;
	private List<EnumeratedDto> accounts;

}

File: fm-domain/src/main/java/io/zeitwert/app/api/rest/dto/UserInfoResponse.kt
package io.zeitwert.app.api.rest.dto

import io.zeitwert.app.api.jsonapi.EnumeratedDto

data class UserInfoResponse(
	val id: Int,
	val name: String,
	val email: String,
	val role: EnumeratedDto,
	val tenants: List<EnumeratedDto>,
)
